### DISCUSSION TITLE
SQLDataSetEditor add the eclipse IDE sreach and replace editor

### DISCUSSION THREAD

**Discussion by speckyspooky** [eclipse-birt/birt#1584] [Q&A]
I started a reserach to add a serach dialoge to the SQLDataSet-Editor for SQL.
The reason is that I have lot of statements which huge lines of SQL-source code (> 1.000 lines per statement).

The idea was from my side to use directly the eclipse-serach-and-replace dialoge to the editor which will be created on the class "SQLDataSetEditorPage". But I didn't found any documention how I can add the standard dialoge from the IDE.

The positive thing is that the central method "getFindReplaceTarget()" is given but I have no real idea of the search-dialoge-implementation.

All hints are welcome how the implementation could be done or some good links to a documentation of it.

**Comment by merks** [comment]
The SQLDataSetEditorPage is a WizardPage, so I assume it's not part of any IEditorPart or any IWorkbenchPart for that matter.  I think the find and replace dialog is only designed to work with editors (or workbench parts?), though I'm not 100% sure.

**Comment by hvbtup** [comment]
Personally, I think that we should not waste effort on improving the Data Set Editor (except perhaps the nasty bug that I have to click "OK" twice all the time).
For developing complex SQL statements, there are much better tools which BIRT cannot compete with.
For example, I am using BIRT for developing reports for an Oracle database.
Since the Oracle JDBC driver supports the `:pi_name` syntax as well as the standard syntax `?` for bind variables, I can just copy and paste the statement from and to Oracle SQL*Developer and develop/test the statement there.

**Reply by speckyspooky** [reply]
I confirm in the most cases I work with the sql developer or the MS SQL Studio.
But sometimes for small changes or error tracking I work directly in the editor of the dataset and then it would be nice to have a search function to get faster the correct position.
